### PR TITLE
provides suggestion for variable name documentation

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

File: go/variable_names.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+### Custom Resource
+
+In our `operatorkit` CRUD resource implementations, we take the parameter `obj`
+and have to assert the proper type the resource actually works with. The
+variable name we want to use here is `customResource`.
+
+```go
+func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
+	customResource, err := key.ToCustomResource(obj)
```
Comment (kopiczko) [inline go/variable_names.md]:
```suggestion
	cr, err := key.ToCustomResource(obj)
```

File: go/variable_names.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+Here we want to provide suggestions and best practises about variable names we
+tend to use in our code. Aligning with the suggestions below makes our code
+usually easier to scan and code in general easier to write because it is a
+no-brainer how to call things, so we can concentrate on what actually matters.
+
+### Custom Resource
+
+In our `operatorkit` CRUD resource implementations, we take the parameter `obj`
+and have to assert the proper type the resource actually works with. The
+variable name we want to use here is `customResource`.
```
Comment (kopiczko) [inline go/variable_names.md]:
```suggestion
variable name we want to use here is `cr`.
```

**Review by kopiczko — COMMENTED** [review]
I'd prefer to rename `customResource` -> `cr`. We use it all over the place. Also when we communicate we always write CR. I think for commonly used vars it makes sense to use short names.

**Comment by xh3b4sd** [comment]
Looking for a couple of approvals here.

File: go/variable_names.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Variable Names
+
+Here we want to provide suggestions and best practises about variable names we
+tend to use in our code. Aligning with the suggestions below makes our code
+usually easier to scan and code in general easier to write because it is a
+no-brainer how to call things, so we can concentrate on what actually matters.
+
+### Custom Resource
+
+In our `operatorkit` CRUD resource implementations, we take the parameter `obj`
```
Comment (kopiczko) [inline go/variable_names.md]: This applies to all resources.

```suggestion
In our `operatorkit` resource implementations, we take the parameter `obj`
```

File: go/variable_names.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+and have to assert the proper type the resource actually works with. The
+variable name we want to use here is `cr`.
+
+```go
+func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
```
Comment (kopiczko) [inline go/variable_names.md]: And the more general method. This is what calls CRUDResource methods.

```suggestion
func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
```

**Review by kopiczko — APPROVED** [review]
Minor suggestions. I like the page in general.

File: go/variable_names.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+### Custom Resource
+
+In our `operatorkit` resource implementations, we take the parameter `obj`
+and have to assert the proper type the resource actually works with. The
+variable name we want to use here is `cr`.
+
+```go
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCustomResource(obj)
```
Comment (rossf7) [inline go/variable_names.md]: <3

File: go/variable_names.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
+# Variable Names
+
+Here we want to provide suggestions and best practises about variable names we
+tend to use in our code. Aligning with the suggestions below makes our code
+usually easier to scan and code in general easier to write because it is a
```
Comment (rossf7) [inline go/variable_names.md]:
```suggestion
easier to scan and code in general easier to write because it is a
```
Flows better for me without.

**Review by rossf7 — APPROVED** [review]
Nice, thanks for adding. ❤️

**Review by MarcelMue — APPROVED** [review]

**Comment by kopiczko** [comment]
Just saying. I think this hangs long enough to be merged. I think ppl had time to express opinions.

**Comment by xh3b4sd** [comment]
Aye aye matey, batten down the hatches.

**Pull request merged**